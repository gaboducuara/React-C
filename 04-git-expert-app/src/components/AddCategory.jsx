import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {

    const [InputValue, setInputValue] = useState('');

    const ChangeInput = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = ( event ) => {
        event.preventDefault();
        if(InputValue.trim().length <= 1) return;

        // setCategories(categories => [InputValue, ...categories]);
        setInputValue('');
        onNewCategory( InputValue.trim() );        
    }

  return (
    <form onSubmit={ onSubmit }>
        <input 
            type="text"
            placeholder='Buscar gifs'
            value={ InputValue }
            onChange={ChangeInput }
        />
    </form>
  );
};