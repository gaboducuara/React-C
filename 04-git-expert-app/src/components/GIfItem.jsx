import PropTypes from 'prop-types'

export const Gifitem = ({ title, url , id }) => {
    
    // console.log({ title, url , id })

  return (
    <div className="card">
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
    )
}

// LOS PROPTYPES
Gifitem.propTypes = {
  title: PropTypes.string.isRequired,
  url:PropTypes.string.isRequired,
}

// Gifitem.defaultProps = {
//   title:'No hay Subtitulo',
//   url:'no existe url',
// }

/* Tarea 

  1.AÃ±adir PropTypes 
    a.title obligatorio
    b. url obligatorio
  
  2.Evaluar snapshot.
*/
