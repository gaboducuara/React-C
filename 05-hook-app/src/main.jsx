import React from 'react';
import ReactDOM from 'react-dom/client';
import './index.css';

// import { Padre } from './07-tarea-memo/Padre';
// import './08-useReducer/intro_reducer'

// import { CallbackHook } from './06-memos/CallbackHook';
// import { MemoHook } from './06-memos/MemoHook'
// import { Memorize } from './06-memos/Memorize'
// import { Layout } from './05-useLayouEffect/Layout'
// import { FocusScreen } from './04-useRef/FocusScreen'
// import { MultipleCustomHook } from './03-examples/MultipleCustomHook'
//import { FormWithCustomHooks } from './02-useEffect/FormWithCustomHook'
// import { SimpleForm } from './02-useEffect/SimpleForm'
// import { CounterWithCustomHooks } from './01-useState/CounterWithCustomHooks'
// import { HooksApp } from './HooksApp'
// import { CounterApp } from './01-useState/CounterApp'
import { TodoApp } from './08-useReducer/TodoApp';

ReactDOM.createRoot(document.getElementById('root')).render(
  //<React.StrictMode>
    <TodoApp />
  //</React.StrictMode>, 
   ) 
