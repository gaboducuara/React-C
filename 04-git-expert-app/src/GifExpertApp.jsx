import { useState } from "react";
import { GifGrid , AddCategory } from "./components";

export const GifExpertApp = () => {
  const [Categories, setCategories] = useState(['saitama']);

  const onAddCategory = (newCategory) => {
    if (Categories.includes(newCategory)) return;

    console.log(newCategory);
    // Categories.push( newCategory )
    setCategories([newCategory, ...Categories]);
  };

  return (
    <>
      <h1> GifExpertApp </h1>
      <AddCategory onNewCategory={(event) => onAddCategory(event)} />
        {
          Categories.map((category) => (
            <GifGrid key={category} category={category} />
          ))
        }
    </>
  );
};