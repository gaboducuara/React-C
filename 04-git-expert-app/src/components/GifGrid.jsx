import PropTypes from "prop-types";
// import { useEffect, useState } from "react";
import { Gifitem } from "./GIfItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
// import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

        {
            isLoading && <h2>Cargando...</h2>
        }

      <div className="card-grid">
        {images.map((image) => (
          <Gifitem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
