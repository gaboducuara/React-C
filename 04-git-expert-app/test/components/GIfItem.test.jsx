import { render, screen} from "@testing-library/react";
import { Gifitem } from "../../src/components";

describe('pruebas en <GIf Item/> en components', () => {

    const title = 'Goku';
    const url = 'http://goku.com/';

    test('debe de hacer match con el snapshot', () => {

        const { container } = render(<Gifitem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();
    });

    test ('debe de mostar la imagen con la URL Y con el ALT indicado' , () => {
        render(<Gifitem title={ title } url={ url } />);
        //screen.debug();
        const {src , alt } = screen.getByRole('presentation');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test ('debe de mostrar el titulo en el componente' , () => {
        render(<Gifitem title={ title } url={ url } />);
        expect(screen.getByText( title )).toBeTruthy();
    })


});