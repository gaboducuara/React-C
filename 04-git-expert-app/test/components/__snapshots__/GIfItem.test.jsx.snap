// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GIf Item/> en components debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="http://goku.com/"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
